postgresql-11.5/src/bin/psql/commandbteq.c

```diff
@@ -345,43 +345,29 @@ static dotResult
 exec_command_set(BteqScanState scan_state, bool active_branch)
 {
     bool        success = true;
-
+    char        quote;
     if (active_branch)
     {
-        char       *opt0 = bteq_scan_dot_option(scan_state,
-                                                  OT_BTEQ_NORMAL, NULL, false);
-
-        if (!opt0)
+        char *opt0 = read_connect_arg(scan_state);
```

Comment:
What does this function do?

Reply:
this function takes the arguments that come after "SET" and checks them

Reply:
if you want to use `read_connect_arg function`, you must need to update comments related to it, and also pass `BTEQ_NORMAL` as an argument to the `bteq_scan_dot_option` inside the function.

Or it is more preferable to define another function similar to this one, which should be used for parsing BTEQ commands, and not only `\connect` command, as it is supposed to do in PSQL source code.

The new function name can be `read_arg`, `read_option` or something else you want.
![image](https://user-images.githubusercontent.com/15236153/67841024-989d3980-fb10-11e9-9b12-9d053a250a2c.png)